### ISSUE TITLE
What is the loss mask for special tokens in SFFTrainer

### ISSUE THREAD

**Issue by RonanKMcGovern** [huggingface/transformers#26620]
### System Info

latest transformers

### Who can help?

@muellerzr and @pacman100

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

I'm training with SFTTrainer and want to ensure that the model is including the loss on predicting an EOS token (< /s >).

What is the default handling of special tokens for the loss computation in SFTTrainer? Can I change this?
```
from transformers import Trainer
from trl import SFTTrainer

trainer = SFTTrainer(
    peft_config=config,
    dataset_text_field="text",
    max_seq_length=context_length,
    tokenizer=tokenizer,
    model=model,
    train_dataset=data["train"],
    eval_dataset=data["test"],
    args=transformers.TrainingArguments(
        max_steps=60, # comment this out after the first time you run. This is for testing!
        num_train_epochs=epochs,
        output_dir=save_dir,
        evaluation_strategy="steps",
        do_eval=True,
        per_device_train_batch_size=batch_size,
        gradient_accumulation_steps=4,
        per_device_eval_batch_size=batch_size,
        log_level="debug",
        optim="paged_adamw_8bit",
        save_steps=0.2,
        logging_steps=1,
        learning_rate=1e-4,
        eval_steps=0.2,
        fp16=True,
        max_grad_norm=0.3,
        warmup_ratio=0.03,
        lr_scheduler_type="linear",
    ),
    callbacks=[logging_callback],  # Add custom callback here
)
model.config.use_cache = False  # silence the warnings. Please re-enable for inference!
trainer.train()
```
Note that in my dataset I have included EOS tokens where appropriate

### Expected behavior

The output of my fine-tuning is not emitting EOS tokens, which leads me to believe that the loss mask is zero for special tokens with SFTTrainer, but I'm unsure if that's true.

**Comment by LysandreJik** [comment]
`SFTTrainer` comes from the `trl` library, let me move this issue there.
